server/src/test/java/com/broadcom/lsp/cobol/usecases/engine/UseCasePreprocessorListener.java

```diff
@@ -477,7 +477,9 @@ private void registerDiagnostics(Range range, List<DiagnosticContext> diagnostic
             .get(documentUri)
             .add(
                 new Diagnostic(
-                    range, it.getMessage(), it.getSeverity(), it.getSource(), it.getCode()));
+                        // honour the range provided by tester.
```

Comment:
Controversial change. I will work on this if the team feels this is not the right approach.
Basically, had to add this as I was unable to add diagnostics across a set of identifiers( which are clubbed together as a single token rule) .